odoo/tools/translate.py

```diff
@@ -1568,11 +1568,13 @@ def _get_translation_upgrade_queries(cr, field):
               GROUP BY res_id
             )
             UPDATE {Model._table} m
-               SET "{field.name}" =  t.value || m."{field.name}"
+               SET "{field.name}" = CASE WHEN imd.noupdate IS TRUE THEN m."{field.name}" || t.value ELSE t.value || m."{field.name}" END
               FROM t
+         LEFT JOIN ir_model_data imd
+                ON imd.res_id = t.res_id AND imd.model = %s AND imd.noupdate IS TRUE
```

Comment:
If you have multiple xmlids pointing to the same record, some noupdate=true, some noupdate=false, what is the expected behavior?

It's probably safer to use a `where not exists` clause instead of a left join. And you can also use 2 queries if you need to update the noupdate ones with a `where exists` clause.

Reply:
> If you have multiple xmlids pointing to the same record, some noupdate=true, some noupdate=false, what is the expected behavior?

It will behavior like `noupdate=true`. The result is deterministic and I think it is acceptable since we also have function `get_external_id` which only returns the first xmlid in the database which is indeterministic

The only tricky thing is a record may have multiple xmlids with `noupdate=true`. In that case, the `LEFT JOIN` will cause multiple translation rows for the same one record  to update. According to the Notes of https://www.postgresql.org/docs/current/sql-update.html, we don't know translations in which row will be used to `UPDATE` but it should be safe here since all translations for these rows are the same.

> It's probably safer to use a where not exists clause instead of a left join. And you can also use 2 queries if you need to update the noupdate ones with a where exists clause.

according to my knowledge using a `JOIN` should be faster then a sub-selects

Reply:
JOIN is faster than subselects.
I was not suggesting subselect for perf, but for correctness.

Reply:
What about a `JOIN`, `GROUP BY`, `bool_or`?